integrations/gcp/gcp_core/search/iterators.py

```diff
@@ -15,15 +14,18 @@ async def iterate_per_available_project(
 ) -> ASYNC_GENERATOR_RESYNC_TYPE:
     try:
         async for projects in search_all_projects():
+
             tasks = [
-                execute_with_concurrency_control(
-                    project_dependent_callable, project, *args, **kwargs
-                )
+                project_dependent_callable(project, *args, **kwargs)
                 for project in projects
             ]
+            if not tasks:
+                logger.warning(
+                    f"Searched {len(projects)} projects and found no accessible resources for {kwargs.get('asset_type')}. This may be due to unset permissions or no currently existing projects."
+                )
+                return
+
             async for batch in stream_async_iterators_tasks(*tasks):
                 yield batch
-    except StopAsyncIteration:
-        raise NoProjectsFoundError(
-            "Performed a search_projects actions, and found no accessable Projects. This may be due to unset permissions or no current existing projects."
-        )
+    except Exception as e:
+        logger.exception(f"Error iterating over projects: {e}")
```

Comment:
this can be problematic to just silent errors, as if we failed to bring data for some kind and the resync didn't receive it as failed it means that the resync passed successfuly and data that it wasn't able to fetch will be deleted as part of the resync process. This is not something we want to do as we would rather not delete from port if the state is incomplete rather than delete data that failed getting from the third part.

Only reasons to silent errors from third party is 403 or 404 where we either don't have access or there is no such resources to fetch.

Reply:
logger.exception actually throws an the error 🤓